btree/btree.c

```diff
@@ -2028,13 +2097,29 @@ static int64_t mem_node_get(struct node_op *op, struct td *tree,
 
 static void mem_node_put(struct nd *node)
 {
-	/**
-	 * This implementation does not perform any action, but the final one
-	 * should decrement the reference count of the node and if this
-	 * reference count goes to '0' then detach this node descriptor from the
-	 * existing node and make this node descriptor available for 'node_get'.
-	 */
-	node->n_ref--;
+	node_refcnt_update(node, false);
+	if (node->n_ref == 0) {
```

Comment:
Let's consider scenario: there are 2 threads refering to same node, therefore n_ref ==2, now both executed line 2100 one after other and now both will reach node->n_ref = 0, and execute if part which should not happen

Reply:
Modified code to handle race condition.